reactor-core/src/main/java/reactor/core/publisher/FluxReplay.java

```diff
@@ -994,17 +999,24 @@ public int size() {
 
 	FluxReplay(CorePublisher<T> source,
 			int history,
-			long ttl,
+			Duration ttl,
 			@Nullable Scheduler scheduler) {
 		this.source = Objects.requireNonNull(source, "source");
 		this.history = history;
 		if(history < 0){
 			throw new IllegalArgumentException("History cannot be negative : " + history);
 		}
-		if (scheduler != null && ttl < 0) {
+		if (scheduler != null && ttl.isNegative()) {
 			throw new IllegalArgumentException("TTL cannot be negative : " + ttl);
 		}
-		this.ttl = ttl;
+		if (Operators.nanoPrecision(ttl)) {
+			this.ttl = ttl.toNanos();
+			this.timeUnit = TimeUnit.NANOSECONDS;
+		}
+		else {
+			this.ttl = ttl.toMillis();
+			this.timeUnit = TimeUnit.MILLISECONDS;
+		}
```

Comment:
The precision logic for FluxReplay can be moved into the constructor of the inner class `SizeAndTimeBoundReplayBuffer`; however, `ReplayProcessor.createSizeAndTimeout` uses this class as well, and it is explicitly documented to use milliseconds.